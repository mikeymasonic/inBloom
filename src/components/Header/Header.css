a {
    text-decoration: none;
  }
  
  a:link {
    color: rgb(151, 40, 40);
  }
  
  a:visited {
    color: rgb(145, 24, 24);
  }
  
  a:hover {
    color: hotpink;
  }
  
  a:active {
    color: hotpink;
  }

.Header {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 0.8fr;
    gap: 1px 1px;
    grid-template-areas: "logo User navbar";
    background-color: rgba(133, 32, 119, 0.658);
    color: white;
    width: 100%;
    height: 6rem;
    margin-bottom: 2rem;
    justify-items: center;

  .nameLink {
    font-size: 1.5rem;
    font-family: var(--display-font);
    margin-right: 2rem;
  }
  
  .logo { 
    grid-area: logo;
    width: 18rem;
    margin-right: 1rem;
    margin-top: 1.5rem;
    margin-bottom: .5rem;
    margin-left: .5rem;
   }
  
  .navbar { 
    grid-area: navbar;
    margin-top: 1rem;
   }
  
  .User { 
    grid-area: User;
    margin-top: 1rem;
    margin-bottom: -1rem;
    margin-left: -8rem;
    line-height: .1rem;
   }

  .navLink {
    margin-right: 1rem;
  }

  .logout {
    font-size: .5rem;
    margin-bottom: 10rem;
  }

  h4 {
    font-size: .75rem;
    margin-top: 1rem;
  }
}
