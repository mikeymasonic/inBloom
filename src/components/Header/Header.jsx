import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './Header.css';
import Player from './../Player/Player';
import Playlist from './../Playlist/Playlist';

const Header = (props) => {
  const [back, setBack] = useState(false);
  const pathname = props.location.pathname;

  useEffect(() => {
    if (pathname != '/') {
      setBack(true);
    } else {
      setBack(false);
    }
  });

  return (
    <header className={styles.Header}>
      <Link className={styles.navLink} to="/">
        <h1 className={styles.logo}>sonic blooming</h1>
      </Link>
      <Player />
      <Playlist />
      <section className={styles.navbar}>
        <Link className={styles.navLink} to="/about">
            _ABOUT
        </Link>
      </section>

      {/* <Link className={styles.navLink} to="/upload">
            _UPLOAD
      </Link> */}
      {back && <h4><Link className={styles.navLink} to="/">
            back
      </Link></h4>}       
    </header>
  );
};

Header.propTypes = {
  location: PropTypes.object,
};

export default Header;
