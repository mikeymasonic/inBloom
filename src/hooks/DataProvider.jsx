import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const DataContext = createContext();

export const DataProvider = ({ children }) => {
  // const [song, setSong] = useState('../mp3/example2.mp3');
  const [song, setSong] = useState(null);
  const [mapLocation, setMapLocation] = useState(null);
  const [playerVisible, setPlayerVisible] = useState(false);
  const [visualizerDisplay, setVisualizerDisplay] = useState(false);

  const handleSong = (track) => {
    setSong(track);
    setMapLocation(track.mapLocation);
  };

  const handleMapLocation = (location) => {
    setMapLocation(location);
  };

  const handlePlayerVisible = (bool) => {
    setPlayerVisible(bool);
  };

  const handleVisualizerDisplay = (bool) => {
    setVisualizerDisplay(bool);
  };

  return (
    <DataContext.Provider
      value={{
        song,
        mapLocation,
        playerVisible,
        visualizerDisplay,
        handleSong,
        handleMapLocation,
        handlePlayerVisible,
        handleVisualizerDisplay,
      }}
    >
      {children}
    </DataContext.Provider>
  );
};

DataProvider.propTypes = {
  children: PropTypes.node,
};

// get state
export const useSong = () => {
  const { song } = useContext(DataContext);
  return song;
};

export const useMapLocation = () => {
  const { mapLocation } = useContext(DataContext);
  return mapLocation;
};

export const usePlayerVisible = () => {
  const { playerVisible } = useContext(DataContext);
  return playerVisible;
};

export const useVisualizerDisplay = () => {
  const { visualizerDisplay } = useContext(DataContext);
  return visualizerDisplay;
};

// get handlers
export const useHandleSong = () => {
  const { handleSong } = useContext(DataContext);
  return handleSong;
};

export const useHandleMapLocation = () => {
  const { handleMapLocation } = useContext(DataContext);
  return handleMapLocation;
};

export const useHandlePlayerVisible = () => {
  const { handlePlayerVisible } = useContext(DataContext);
  return handlePlayerVisible;
};

export const useHandleVisualizerDisplay = () => {
  const { handleVisualizerDisplay } = useContext(DataContext);
  return handleVisualizerDisplay;
};

