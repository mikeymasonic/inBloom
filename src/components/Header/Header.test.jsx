import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { DataProvider } from '../../hooks/DataProvider';
import Header from './Header';

describe('Header component', () => {
  afterEach(() => cleanup());
  xit('renders Header', () => {
    const { asFragment } = render(
      <Router>
        <DataProvider>
          <Header />
        </DataProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
