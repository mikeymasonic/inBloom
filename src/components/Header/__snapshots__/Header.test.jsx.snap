// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <header
    class="navLink"
  >
    <a
      class="navLink"
      href="/"
    >
      <h1>
        sonic bloom
      </h1>
    </a>
    <a
      class="navLink"
      href="/about"
    >
      _ABOUT
    </a>
  </header>
</DocumentFragment>
`;
